include/drivers/qspi.h

```diff
@@ -0,0 +1,660 @@
+				const struct qspi_buf *tx_buf, u32_t address)
+{
+	const struct qspi_driver_api *api =
+		(const struct qspi_driver_api *)dev->driver_api;
+
+	return api->write(dev, tx_buf, address);
+}
+
+
+/**
+ * @brief Read desired amount of data from the external flash memory.
+ *
+ * Note: This function is synchronous.
+ * Driver that realizes read functionality has to:
+ *  1. Check if memory is ready for operation - using RDSR
+ *	   (read status register) command
+ *	2. Send desired read command (depends
+ *	   on how many data lines are used) and address.
+ *
+ *	For full list of supported commands refer to the
+ *	manufacturers datasheet.
+ *	If driver is configured to use four data lines one must
+ *	enable quad transfer for the memory.
+ *	Refer to qspi_configure description.
+ *
+ * @param dev Pointer to the device structure for the driver instance
+ * @param rx_buf Buffer for receive purpose.
+			For detailed description please refer to the qspi_buf
+ * @param address Address from the data will be read.
+ *
+ * @retval  0 in case of success
+ * @retval	-ENXIO No such device or address
+ * @retval	-EINVAL invalid input parameter
+ * @retval	-EBUSY device busy
+ */
+__syscall int qspi_read(struct device *dev,
+				const struct qspi_buf *rx_buf, u32_t address);
+
+static inline int z_impl_qspi_read(struct device *dev,
+				const struct qspi_buf *rx_buf, u32_t address)
+{
+	const struct qspi_driver_api *api =
+		(const struct qspi_driver_api *)dev->driver_api;
+
+	return api->read(dev, rx_buf, address);
+}
+
+
+/**
+ * @brief Function for sending operation code, sending data
+ * and receiving data from the memory device.
+ *
+ * Note: This function is synchronous.
+ * For command issuing driver uses only one data line.
+ * This is universal function, so user can send
+ * i.e read/write/erase command using this command.
+ * If user would like to send command with address and
+ * payload (i.e write command), one must
+ * fill op_code(i.e PP - Page Program - 0x02)
+ * and tx_buff (address and data).
+ * For command structure refer to the memory manufacturers datasheet.
+ *
+ * Driver that realizes custom command functionality has to:
+ *  1. Check if memory is ready for operation - using RDSR
+ *	   (read status register) command
+ *
+ * @param dev Pointer to the device structure for the driver instance
+ * @param cmd Command structure.
+ * For detailed description please refer to the qspi_cmd
+ *
+ * @retval  0 in case of success
+ * @retval	-ENXIO No such device or address
+ * @retval	-EINVAL invalid input parameter
+ * @retval	-EBUSY device busy
+ */
+__syscall int qspi_send_cmd(struct device *dev, const struct qspi_cmd *cmd);
+
+static inline int z_impl_qspi_send_cmd(struct device *dev,
+				const struct qspi_cmd *cmd)
+{
+	const struct qspi_driver_api *api =
+		(const struct qspi_driver_api *)dev->driver_api;
+
+	return api->send_cmd(dev, cmd);
+}
+
+/**
+ * @brief Erases desired amount of flash memory.
```

Comment:
This is a feature of JEDEC, not of QSPI.  Should this be in the JEDEC driver that calls this?